node/src/components/consensus/candidate_block.rs

```diff
@@ -85,4 +78,12 @@ impl ConsensusValueT for CandidateBlock {
     fn needs_validation(&self) -> bool {
         !self.proto_block.wasm_deploys().is_empty() || !self.proto_block.transfers().is_empty()
     }
+
+    fn timestamp(&self) -> Timestamp {
+        self.proto_block.timestamp()
```

Comment:
Note: timestamp has moved into `ProtoBlock`, that is why it is missing from all of the arguments.

TBD: move parent into `ProtoBlock` as well.

Reply:
…or move timestamp out again, as discussed. :sweat_smile: